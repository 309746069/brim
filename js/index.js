import React from "react"
import ReactDOM from "react-dom"
import {AppContainer} from "react-hot-loader"
import {Provider} from "react-redux"
import initStore from "./initStore"
import {HashRouter} from "react-router-dom"
import XApp from "./connectors/XApp"
const store = initStore()

ReactDOM.render(
  <AppContainer>
    <Provider store={store}>
      <HashRouter>
        <XApp />
      </HashRouter>
    </Provider>
  </AppContainer>,
  document.getElementById("app-root")
)
