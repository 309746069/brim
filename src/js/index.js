import React from "react"
import ReactDOM from "react-dom"
import {Provider} from "react-redux"
import initStore from "./initStore"
import {HashRouter} from "react-router-dom"
import XApp from "./connectors/XApp"
const store = initStore()

ReactDOM.render(
  <Provider store={store}>
    <HashRouter>
      <XApp />
    </HashRouter>
  </Provider>,
  document.getElementById("app-root")
)
