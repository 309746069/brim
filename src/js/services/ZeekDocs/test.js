/* @flow */
import ZeekDocs from "./"

test("describe conn uid", () => {
  let desc = ZeekDocs.describe("conn", "uid")

  expect(desc).toBe("A unique identifier of the connection.")
})
