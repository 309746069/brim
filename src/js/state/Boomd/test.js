/* @flow */
import Boomd from "./"
import initTestStore from "../../test/initTestStore"

test("setting use analytics cache", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useCache(true))
  expect(Boomd.usingCache(store.getState())).toEqual(true)
  store.dispatch(Boomd.useCache(false))
  expect(Boomd.usingCache(store.getState())).toEqual(false)
})

test("setting use index", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useIndex(true))
  expect(Boomd.usingIndex(store.getState())).toEqual(true)
  store.dispatch(Boomd.useIndex(false))
  expect(Boomd.usingIndex(store.getState())).toEqual(false)
})
