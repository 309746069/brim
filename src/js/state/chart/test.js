/* @flow */

import chart from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let records = [
  [{name: "id", type: "string", value: "1"}],
  [{name: "id", type: "string", value: "2"}]
]

test("chart records append", () => {
  let state = store.dispatchAll([
    chart.appendRecords([records[0]]),
    chart.appendRecords([records[1]])
  ])

  expect(chart.getRecords(state)).toEqual(records)
})

test("chart records status", () => {
  let state = store.dispatchAll([chart.setStatus("SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  let state = store.dispatchAll([
    chart.appendRecords(records),
    chart.setStatus("SUCCESS"),
    chart.clear()
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getRecords(state)).toEqual([])
})
