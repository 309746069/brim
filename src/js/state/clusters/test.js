/* @flow */

import {addCluster, removeCluster, setCluster, setClusterError} from "./actions"
import {
  getCurrentCluster,
  getClusterMessage,
  getSavedClusters
} from "./selectors"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let cluster = {
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  let state = store.dispatchAll([addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("addCluster when it already exists", () => {
  let state = store.dispatchAll([addCluster(cluster), addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("removeCluster", () => {
  let state = store.dispatchAll([addCluster(cluster), removeCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([])
})

test("setCluster", () => {
  let state = store.dispatchAll([setCluster(cluster)])

  expect(getCurrentCluster(state)).toEqual(cluster)
})

test("setClusterMessage to message", () => {
  let state = store.dispatchAll([setClusterError("no spaces")])

  expect(getClusterMessage(state)).toEqual("no spaces")
})

test("setClusterMessage to empty", () => {
  let state = store.dispatchAll([setClusterError("")])

  expect(getClusterMessage(state)).toEqual("")
})
