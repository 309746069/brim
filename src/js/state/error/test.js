/* @flow */
import AppError from "../../models/AppError"
import error from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("init state", () => {
  let state = store.getState()
  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(false)
})

test("set", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e)])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(true)
})

test("clear", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.clear()])

  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(true)
})

test("dismiss", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.dismiss()])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(false)
})
