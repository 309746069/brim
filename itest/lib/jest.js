/* @flow */

import {Application} from "spectron"

import {selectors} from "../../src/js/test/integration"

export const TestTimeout = 60000

// This is trying to make Flow happy for passing in the done function from
// Jest.
export type Done = {
  (string | Error | void): void,
  fail?: (string | Error | void) => void
}

export const handleError = async (
  app: Application,
  initialError: Error,
  done: Done
) => {
  let realError = undefined
  let notificationError = undefined
  console.log(`handleError: Test hit exception: ${initialError.message}`)
  console.log("handleError: Looking for any desktop app notifications")
  try {
    notificationError = await app.client.getHTML(selectors.notification, false)
  } catch (e) {
    notificationError = undefined
  }
  if (notificationError) {
    realError = new Error(
      "App notification '" +
        notificationError +
        "' (initial error: '" +
        initialError +
        "'"
    )
  } else {
    console.log("handleError: desktop app notification not found")
    realError = initialError
  }
  done.fail && done.fail(realError)
}
