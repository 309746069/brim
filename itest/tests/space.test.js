/* @flow */

import {basename} from "path"

import {
  getCurrentSpace,
  logIn,
  newAppInstance,
  resetState,
  searchDisplay,
  setSpace,
  setSpan,
  startApp,
  startSearch,
  waitForSearch,
  writeSearch
} from "../lib/app.js"
import {handleError, stdTest} from "../lib/jest.js"
import {dataSets} from "../../src/js/test/integration"

describe("Spaces tests", () => {
  let app
  let testIdx = 0
  beforeEach(() => {
    app = newAppInstance(basename(__filename), ++testIdx)
    return startApp(app)
  })

  afterEach(async () => {
    if (app && app.isRunning()) {
      await resetState(app)
      return await app.stop()
    }
  })

  stdTest(`default space is ${dataSets.hq_integration.spaceName}`, (done) => {
    logIn(app)
      .then(() => getCurrentSpace(app))
      .then((spaceName) => {
        expect(spaceName).toBe(dataSets.corelight.spaceName)
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })

  stdTest(`change space to ${dataSets.hq_integration.spaceName}`, (done) => {
    logIn(app)
      .then(() => setSpace(app, dataSets.hq_integration.spaceName))
      .then(() => getCurrentSpace(app))
      .then((spaceName) => {
        expect(spaceName).toBe(dataSets.hq_integration.spaceName)
      })
      .then(() => writeSearch(app, "* | count()"))
      .then(() => startSearch(app))
      .then(() => waitForSearch(app))
      .then(() => searchDisplay(app))
      .then((results) => {
        expect(results).toMatchSnapshot()
      })
      .then(() => setSpan(app, "Whole Space"))
      .then(() => waitForSearch(app))
      .then(() => searchDisplay(app))
      .then((results) => {
        expect(results).toMatchSnapshot()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })
})
