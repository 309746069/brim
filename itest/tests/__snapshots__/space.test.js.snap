// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spaces tests change space to hq_integration-zeek-bulk 1`] = `
Array [
  "count",
  "4",
]
`;

exports[`Spaces tests change space to hq_integration-zeek-bulk 2`] = `
Array [
  "count",
  "41,088,058",
]
`;
